### PR TITLE
Virtual keyboard redesign

### PR DESCRIPTION
**Pull request by ElderOrb**
should resolve: 

https://highfidelity.fogbugz.com/f/cases/10605/crash-in-setKeyboardRaised-when-oking-or-canceling-asset-import-dialog

https://highfidelity.fogbugz.com/f/cases/13709/CRASH-Attempting-to-upload-files-via-URL-causes-Interface-to-crash

Ongoing work on redesign keyboard infrastructure with the following goals: 

1. As less intrusive as possible. Ideally client code should be left untouched
2. C++ part should never operate 'dead' items
3. New implementation should allow to use self-destructable dialog

Current PR introduces early POC on single-per-surface, never destructable keyboard

Open issues: 
1. How to properly find *right* keyboard?
_At the moment the only solution I see is just specify keyboard using separate property -  'keyboardContainer'. Fortunately it should be required only in a few cases. One of which is Wallet_ 

Update: after discussion with Austin and Dante all the raise/lower keyboard triggers where moved to QML side.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8623/app_12834_PR.apk

**Comment by ElderOrb** [comment]
[HifiOnscreenKeyboardPrototype.zip](https://github.com/highfidelity/hifi/files/1900108/HifiOnscreenKeyboardPrototype.zip)
Just in case attaching standalone prototype

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8773/app_12834_PR_8d29b3434bbd5b5661aa878116a7e96f3c7268fa.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8773/HighFidelity-Beta-PR12834-8d29b3434bbd5b5661aa878116a7e96f3c7268fa.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8773/HighFidelity-Beta-PR12834-8d29b3434bbd5b5661aa878116a7e96f3c7268fa.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8777/app_12834_PR_4f285f05363423626e37ca65504318cf5ed46b6e.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8777/HighFidelity-Beta-PR12834-4f285f05363423626e37ca65504318cf5ed46b6e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8777/HighFidelity-Beta-PR12834-4f285f05363423626e37ca65504318cf5ed46b6e.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10651/HighFidelity-Beta-PR12834-92412fc.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10651/HighFidelity-Beta-PR12834-92412fc.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10651/HighFidelity-Beta-Interface-PR12834-92412fc.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10651/HighFidelity-Beta-PR12834-92412fc.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10651/HighFidelity-Beta-Interface-PR12834-92412fc.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10652/HighFidelity-Beta-PR12834-42af828.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10652/HighFidelity-Beta-PR12834-42af828.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10652/HighFidelity-Beta-Interface-PR12834-42af828.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10652/HighFidelity-Beta-PR12834-42af828.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10652/HighFidelity-Beta-Interface-PR12834-42af828.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10690/HighFidelity-Beta-PR12834-4ea98e7.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10692/HighFidelity-Beta-PR12834-118c1e5.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10692/HighFidelity-Beta-PR12834-118c1e5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10692/HighFidelity-Beta-Interface-PR12834-118c1e5.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10898/HighFidelity-Beta-PR12834-aa06ace.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10898/HighFidelity-Beta-PR12834-aa06ace.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10898/HighFidelity-Beta-Interface-PR12834-aa06ace.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10898/HighFidelity-Beta-PR12834-aa06ace.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10898/HighFidelity-Beta-Interface-PR12834-aa06ace.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10910/HighFidelity-Beta-PR12834-faa0c94.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10910/HighFidelity-Beta-PR12834-faa0c94.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10910/HighFidelity-Beta-Interface-PR12834-faa0c94.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10910/HighFidelity-Beta-PR12834-faa0c94.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10910/HighFidelity-Beta-Interface-PR12834-faa0c94.exe

File: interface/resources/qml/controls-uit/VirtualKeyboard.qml [hunk 1]
```diff
@@ -0,0 +1,421 @@
+//
+//  FileDialog.qml
```
Comment (jherico) [inline interface/resources/qml/controls-uit/VirtualKeyboard.qml]: fix

File: interface/resources/qml/controls-uit/VirtualKeyboard.qml [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+//
+//  Created by Anthony Thibault on 31 Oct 2016
```
Comment (jherico) [inline interface/resources/qml/controls-uit/VirtualKeyboard.qml]: fix

Comment (ElderOrb) [reply interface/resources/qml/controls-uit/VirtualKeyboard.qml]: done

File: interface/resources/qml/controls-uit/VirtualKeyboard.qml (hunk 1 above)
Comment (ElderOrb) [reply interface/resources/qml/controls-uit/VirtualKeyboard.qml]: done

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10985/HighFidelity-Beta-PR12834-914a5c5.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10985/HighFidelity-Beta-PR12834-914a5c5.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10985/HighFidelity-Beta-Interface-PR12834-914a5c5.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10985/HighFidelity-Beta-PR12834-914a5c5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10985/HighFidelity-Beta-Interface-PR12834-914a5c5.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11014/HighFidelity-Beta-PR12834-dd87dd1.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11014/HighFidelity-Beta-PR12834-dd87dd1.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11014/HighFidelity-Beta-Interface-PR12834-dd87dd1.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11017/HighFidelity-Beta-PR12834-72d5e23.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11017/HighFidelity-Beta-PR12834-72d5e23.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11017/HighFidelity-Beta-Interface-PR12834-72d5e23.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11017/HighFidelity-Beta-PR12834-72d5e23.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11017/HighFidelity-Beta-Interface-PR12834-72d5e23.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11059/HighFidelity-Beta-PR12834-953efe9.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11059/HighFidelity-Beta-PR12834-953efe9.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11059/HighFidelity-Beta-Interface-PR12834-953efe9.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11059/HighFidelity-Beta-PR12834-953efe9.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11059/HighFidelity-Beta-Interface-PR12834-953efe9.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11090/HighFidelity-Beta-PR12834-be55961.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11090/HighFidelity-Beta-PR12834-be55961.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11090/HighFidelity-Beta-Interface-PR12834-be55961.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11090/HighFidelity-Beta-PR12834-be55961.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11090/HighFidelity-Beta-Interface-PR12834-be55961.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11124/HighFidelity-Beta-PR12834-1236038.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11124/HighFidelity-Beta-Interface-PR12834-1236038.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11124/HighFidelity-Beta-PR12834-1236038.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11124/HighFidelity-Beta-Interface-PR12834-1236038.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11140/HighFidelity-Beta-PR12834-2c871b2.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11140/HighFidelity-Beta-PR12834-2c871b2.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11140/HighFidelity-Beta-Interface-PR12834-2c871b2.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11140/HighFidelity-Beta-PR12834-2c871b2.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11140/HighFidelity-Beta-Interface-PR12834-2c871b2.exe

**Comment by ZappoMan** [comment]
super old and out of date

**Pull request closed without merging**